### PR TITLE
Refactor TrackPlaceAction so block brake logic only runs once

### PR DESCRIPTION
**Pull request by spacek531**
In TrackPlaceAction, block brake related functions run for every block in the sequence. TrackRemoveAction does it only after all the blocks are removed. This refactor moves TrackPlaceAction's block brake code to match TrackRemoveAction's so it only runs once. This requires creating a variable to store the position of the sequence 0 TrackElement, which may be useful for future ride features.

This feature will be useful for diagonal block brakes.

### CODE REVIEW

**Comment by Gymnasiast** [comment]
Can you explain a bit more about this - why is it necessary, and what are the caveats/catches with this new setup?

**Comment by spacek531** [comment]
The primary reasons for this are:
- Simplify the code by removing the multiple `trackBlock->index ==/!= 0` checks.
- Prevents having to add yet another check for future multi-tile block brakes (diagonal block brakes).

The side benefits are:
- `TrackPlaceAction` and `TrackRemoveAction` now have the same form where reciprocal code blocks are executed in the same order.
- Any potential bugs that add TileElements that intersect with multi-tile track elements in between query and execute will not erroneously increment the number of block brakes.
- `invalidate_test_results(ride)` is called once per track element instead of once per tile element


For these benefits, TrackPlaceAction must store the position of the zeroth tile element.

File: src/openrct2/actions/TrackPlaceAction.cpp [hunk 1]
```diff
@@ -701,6 +649,55 @@ GameActions::Result TrackPlaceAction::Execute() const
         MapInvalidateTileFull(mapLoc);
     }
 
+    if (!(GetFlags() & GAME_COMMAND_FLAG_GHOST))
```
Comment (duncanspumpkin) [inline src/openrct2/actions/TrackPlaceAction.cpp:652]: Could you add a comment to this block something like "Update ride/track stats and interactions"

**Review by duncanspumpkin — APPROVED** [review]
All looks sensible could you rebase and add the comment

**Pull request merged**